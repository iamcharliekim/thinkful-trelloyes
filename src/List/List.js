import React from 'react'
import './List.css'
import Card from '../Card/Card'

const list = (props) => {
	let header = props.header
	let cards = props.cards
	
	let listCards = cards.map((card, i) => {
		return <Card title = {card.title} content = {card.content} key = {i} deleteHandler = {()=> props.deleteHandler(card.id)}/> 
	})
	return (
		<section className="List">
			<header className="List-header"><h2>{header}</h2></header>
			<div className="List-cards">
				{listCards}
				<button type="button" className="List-add-button" onClick={()=>props.randomCardHandler(props.id)}>
              + Add Random Card
            </button>
			</div>
		</section>
	
	)
}

export default list