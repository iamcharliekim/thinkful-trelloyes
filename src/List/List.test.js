import React from 'react'
import ReactDOM from 'react-dom'
import List from './List'
import renderer from 'react-test-renderer'

describe('List component', ()=>{
	it('renders List without crashing', ()=>{
		let div = document.createElement('div')

		ReactDOM.render(<List header='example header' cards={[{title: 'exampleTitle', content: 'exampleContent'}]}/>, div)

		ReactDOM.unmountComponentAtNode(div)
	})

	it('renders the UI of List as expected', ()=> {
		const tree = renderer.create(<List header='example header' cards={[{title: 'exampleTitle', content: 'exampleContent'}]} />).toJSON();

		expect(tree).toMatchSnapshot()
	})
	
	
})
