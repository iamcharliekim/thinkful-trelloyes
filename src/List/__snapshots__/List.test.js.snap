// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI of List as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      example header
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        exampleTitle
      </h3>
      <p>
        exampleContent
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;

exports[`renders the UI of List as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      example header
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        exampleTitle
      </h3>
      <p>
        exampleContent
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
